src/drafty.js

```diff
@@ -1688,3 +2009,93 @@ function copyLight(ent) {
 if (typeof module != 'undefined') {
   module.exports = Drafty;
 }
+
+/**
+ * Callback for applying custom modifications/transformation to a reply quote preview
+ * styles and entities. It may keep the style/entity unmodified,
+ * change or remove them (return null).
+ * Called once per every formatting entry.
+ * @memberof Drafty
+ * @static
+ *
+ * @callback StyleTransform
+ * @param {Object} style - style object.
+ * @param {Object} entity - entity data corresponding to style (may be null).
+ *
+ * @return Array - a 2-entry array of Object [transformed style, transformed entity].
+ */
+
+/**
+ * Shorten Drafty document and modify formatting and entity data
+ * to be suitable for reply quote previews.
+ * @memberof Drafty
+ * @static
+ *
+ * @param {Drafty} original - Drafty object to shorten.
+ * @param {number} length - length in characters to shorten to.
+ * @param {StyleTransform} transform - style transformation callback. 
+ * @returns new shortened Drafty object leaving the original intact.
+ */
+Drafty.replyPreview = function(original, length, transform) {
```

Comment:
Is it possible to generalize `Drafty.preview` to make it suitable for both? There is a lot of code duplication between `Drafty.preview` and `Drafty.replyPreview`.